src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -329,13 +329,18 @@ public override Task<bool> HandleRequestAsync()
                 RedirectUri = BuildRedirectUri(Options.CallbackPath),
                 Resource = Options.Resource,
                 ResponseType = Options.ResponseType,
-                Prompt = Options.Prompt,
-                Scope = string.Join(" ", Options.Scope)
+                Prompt = RetrieveQueryParameterValue(properties, OpenIdConnectParameterNames.Prompt) ?? Options.Prompt,
```

Comment:
I don't really like how this lives in the handler, this seems more natural as part of the AuthenticationProperties class itself

`Prompt = properties.GetAndRemoveItem(key)`

Reply:
Actually if this is something we are going to support as a widespread pattern, we might as well just add a new parallel Parameters property to AuthenticationProperties, and stop putting things and removing things from the Items.

All of these properties could just target the Parameters dictionary and leave them there, since its a new property, the state formatter won't be affected by these, and we can get rid of this hacky stuff insert and remove logic.

So this would just become `Prompt = properties.Parameters[OpenIdConnectParameterNames.Prompt] ?? Options.Prompt`

Reply:
I agree that this would be cleaner but my concern was that a more widespread change (like changing the base AuthenticationProperties type) would likely prevent this change from landing with 2.1. And I opted for a method in the RemoteAuthenticationHandler instead of an extension method because that way I wouldn’t be introducing yet another class ^^

Also, relying on a separate collection would unfortunately be a breaking change for the GoogleHandler who has supported being configured through the properties Items for a while now. :/

Reply:
So its the scope of the change that puts it at risk for 2.2, as opposed to whether its cross repo or not.  If anything, its more likely to get punted to 2.2 if its not as clean as we'd like and would benefit from more bake time...

Yeah I noticed that Google would have to be handled specially for breaking changes, but I think we should come up with the best pattern for widespread use, and if google has to leave an existing codepath for back compat, in addition to supporting the new pattern, that's much better than spreading a less than ideal pattern everywhere else.  

Currently passing parameters to challenge via properties.Items looks pretty much like a hack, with the insert/remove before state serialization.  If this is going to be adopted across the majority of the providers, its much more important this is done well as opposed to quickly.

That said, I don't think making changes to http abstractions will affect how long this will take. since I'm just proposing a new parallel property for dedicated use for parameter passing, if anything it should be safer since we won't be mucking with the Items collection anymore, so its less likely to have any sideffects.

Reply:
At a minimum make GetAndRemoveItem an independent extension method, not part of remote auth handler. It would likely go somewhere in HttpAbstractions.